### PR TITLE
Fix counts cache update for too few subscribers [MAILPOET-5390]

### PR DESCRIPTION
**Pull request by alex-mailpoet**
## Description

_N/A_

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5390]

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5390]: https://mailpoet.atlassian.net/browse/MAILPOET-5390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Subscribers/SubscribersCountsController.php [hunk 1]
```diff
@@ -169,9 +169,7 @@ private function getCacheItem(string $key, int $id): ?array {
   }
 
   private function setCacheItem(string $key, array $item, int $id): void {
-    if ($this->subscribersFeature->isSubscribersCountEnoughForCache()) {
-      $this->transientCache->setItem($key, $item, $id);
-    }
+    $this->transientCache->setItem($key, $item, $id);
```
Comment (costasovo) [inline mailpoet/lib/Subscribers/SubscribersCountsController.php]: @alex-mailpoet  I see how it helps with the issue. But if I'm not mistaken, this change's side effect is that if the site has a low number of subscribers, the plugin will write to cache (so to DB) on every page load when subscriber count data are needed. It is probably not a heavy query, but also not a nice behavior.

Maybe we could fix it in UI or in the place where we send the data about the cache age to UI, and if `isSubscribersCountEnoughForCache` is `false` we say the cache is up to date and send 0.

What do you think?

**Review by costasovo — COMMENTED** [review]
I added a question. Please see my other comment.

File: mailpoet/lib/Subscribers/SubscribersCountsController.php (hunk 1 above)
Comment (alex-mailpoet) [reply mailpoet/lib/Subscribers/SubscribersCountsController.php]: @costasovo Good idea, Rosta! I have made a change according to your proposition.

**Review by costasovo — APPROVED** [review]
LGTM! 👍

**Pull request merged**